main.go

```diff
@@ -1,96 +1,95 @@
 package main
 
 import (
-	"crypto/md5"
-	"errors"
 	"fmt"
-	"io"
 	"os"
-	"strings"
 
-	"github.com/bitrise-io/go-android/cache"
+	"github.com/bitrise-io/go-android/v2/cache"
 	utilscache "github.com/bitrise-io/go-steputils/cache"
-	"github.com/bitrise-io/go-utils/command"
-	"github.com/bitrise-io/go-utils/log"
-	"github.com/bitrise-io/go-utils/pathutil"
-	"github.com/bitrise-io/go-steputils/input"
-	shellquote "github.com/kballard/go-shellquote"
+	"github.com/bitrise-io/go-steputils/v2/export"
+	"github.com/bitrise-io/go-steputils/v2/stepconf"
+	"github.com/bitrise-io/go-utils/v2/command"
+	"github.com/bitrise-io/go-utils/v2/env"
+	"github.com/bitrise-io/go-utils/v2/log"
+	"github.com/bitrise-io/go-utils/v2/pathutil"
+	"github.com/kballard/go-shellquote"
 )
 
-// ConfigsModel ...
-type ConfigsModel struct {
-	GradleFile    string
-	UnitTestTasks string
-	GradlewPath   string
-	UnitTestFlags string
-	DeployDir     string
-	CacheLevel    string
+type Inputs struct {
+	GradleFile    string `env:"gradle_file"`
+	UnitTestTasks string `env:"unit_test_task,required"`
+	GradlewPath   string `env:"gradlew_file_path,required"`
+	UnitTestFlags string `env:"unit_test_flags"`
+	DeployDir     string `env:"BITRISE_DEPLOY_DIR"`
+	CacheLevel    string `env:"cache_level,opt[all,only deps,none]"`
 }
 
-func createConfigsModelFromEnvs() ConfigsModel {
-	return ConfigsModel{
-		GradleFile:    os.Getenv("gradle_file"),
-		UnitTestTasks: os.Getenv("unit_test_task"),
-		GradlewPath:   os.Getenv("gradlew_file_path"),
-		UnitTestFlags: os.Getenv("unit_test_flags"),
-		DeployDir:     os.Getenv("BITRISE_DEPLOY_DIR"),
-		CacheLevel:    os.Getenv("cache_level"),
-	}
-}
+func main() {
+	// Setup dependencies
+	logger := log.NewLogger()
+	envRepo := env.NewRepository()
+	stepInputParser := stepconf.NewInputParser(envRepo)
+	pathChecker := pathutil.NewPathChecker()
+	cmdFactory := command.NewFactory(envRepo)
+	outputExporter := export.NewExporter(cmdFactory)
 
-func (configs ConfigsModel) print() {
-	log.Infof("Configs:")
-	log.Printf("- GradleFile: %s", configs.GradleFile)
-	log.Printf("- UnitTestTasks: %s", configs.UnitTestTasks)
-	log.Printf("- GradlewPath: %s", configs.GradlewPath)
-	log.Printf("- UnitTestFlags: %s", configs.UnitTestFlags)
+	// Parse inputs
+	var inputs Inputs
+	if err := stepInputParser.Parse(&inputs); err != nil {
+		failF(logger, "issue with input: %s", err)
+	}
 
-	log.Printf("- DeployDir: %s", configs.DeployDir)
-	log.Printf("- CacheLevel: %s", configs.CacheLevel)
-}
+	stepconf.Print(inputs)
+	logger.Println()
 
-func (configs ConfigsModel) validate() (string, error) {
```

Comment:
Function renamed to validateInputs and only the optional GradleFile validation is kept, the rest is migrated to use the stepconf package